velox/exec/tests/TableWriteTest.cpp

```diff
@@ -610,3 +644,109 @@ TEST_F(TableWriteTest, writeNoFile) {
   execute(plan, std::make_shared<core::QueryCtx>(executor_.get()));
   ASSERT_TRUE(fs::is_empty(outputDirectory->path));
 }
+
+TEST_F(TableWriteTest, insertUnpartitionedTableWithExistingTableType) {
+  // Inserts into existing unpartitioned tables with "immutable_partitions" set
+  // to false are not allowed. We assert that an error is thrown.
+
+  auto input = makeVectors(rowType_, 10, 10);
+  auto outputDirectory = TempDirectoryPath::create();
+  auto plan = createInsertPlan(
+      input,
+      rowType_,
+      connector::hive::LocationHandle::TableType::kExisting,
+      outputDirectory->path);
+
+  VELOX_ASSERT_THROW(
+      AssertQueryBuilder(plan)
+          .connectorConfig(
+              kHiveConnectorId,
+              HiveConfig::kImmutablePartitions,
+              folly::to<std::string>("true"))
+          .copyResults(pool()),
+      "Unpartitioned Hive tables are immutable.");
+}
+
+TEST_F(TableWriteTest, insertUnpartitionedTableWithNewTableType) {
+  // When table type is NEW, we always return UpdateMode::kNew. In this case
+  // no exception is expected because we are trying to insert rows into the
+  // table.
+
+  for (auto immutablePartitionsEnabled : {"true", "false"}) {
+    auto input = makeVectors(rowType_, 10, 10);
+    auto outputDirectory = TempDirectoryPath::create();
+    auto plan = createInsertPlan(
+        input,
+        rowType_,
+        connector::hive::LocationHandle::TableType::kNew,
+        outputDirectory->path);
+
+    auto result = AssertQueryBuilder(plan)
+                      .connectorConfig(
+                          kHiveConnectorId,
+                          HiveConfig::kImmutablePartitions,
+                          folly::to<std::string>(immutablePartitionsEnabled))
+                      .copyResults(pool());
+
+    assertEqualResults(
+        {makeRowVector({makeConstant<int64_t>(100, 1)})}, {result});
+  }
+}
+
+TEST_F(TableWriteTest, insertUnpartitionedTable) {
+  // This test uses the default value "false" for the "immutable_partitions"
+  // config allowing writing to unpartitioned tables.
+  for (auto numRows : {0, 10, 100}) {
+    auto input = makeVectors(rowType_, 10, numRows);
+
+    createDuckDbTable(input);
+
+    for (auto tableType : tableTypes) {
+      auto outputDirectory = TempDirectoryPath::create();
+      auto plan =
+          createInsertPlan(input, rowType_, tableType, outputDirectory->path);
+      assertQuery(plan, "SELECT count(*) from tmp");
+      assertQuery(
+          PlanBuilder().tableScan(rowType_).planNode(),
+          makeHiveConnectorSplits(outputDirectory),
+          "SELECT * FROM tmp");
+    }
+  }
+}
+
+TEST_F(TableWriteTest, insertUnpartitionedTableSequence) {
```

Comment:
Is this a test for "append" behavior?

Reply:
Yes. All tests that are added in this PR verify Append behaviour.